src/worker_pool/base.ts

```diff
@@ -0,0 +1,106 @@
+import assert from 'node:assert';
+
+export abstract class AsynchronouslyCreatedResource {
+    onreadyListeners : (() => void)[] | null = [];
+
+    markAsReady () : void {
+      const listeners = this.onreadyListeners;
+      assert(listeners !== null);
+      this.onreadyListeners = null;
+      for (const listener of listeners) {
+        listener();
+      }
+    }
+
+    isReady () : boolean {
+      return this.onreadyListeners === null;
+    }
+
+    onReady (fn : () => void) {
+      if (this.onreadyListeners === null) {
+        fn(); // Zalgo is okay here.
```

Comment:
```suggestion
        fn(); // algo is okay here.
```

Reply:
Its different, the term [Zalgo](https://progressivecoder.substack.com/p/dont-unleash-zalgo-in-your-nodejs) applies here.